api/deploy/ocp/go.mod

```diff
@@ -0,0 +1,9 @@
+module github.com/ceph/ceph-csi/api/deploy/ocp
```

Comment:
@nixpanic we will have one more go.mod for the kubernetes specific folder also? instead of that is it makes it easy to have single go.mod for api's?

Reply:
Yes, it is intentional to have `go.mod` for a Kubernetes package too. If there would be an other container platform that we want to support this way, it will prevent pulling in dependencies from different platforms.

But, on the other hand, golang might be smart enough to include only the dependencies that are needed. So, I don't really have a strong preference.

Reply:
For easy maintenance  IMO we  can have single go.mod in api/deploy folder

Reply:
sure, I'll move it around.